jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyManagerTest.java

```diff
@@ -13,208 +13,213 @@
-    BuildConfiguration buildConfig = createBuildConfig().toBuilder()
-        .assembly(AssemblyConfiguration.builder().name("other-layer").build())
-        .build();
-
-    AssemblyManager.verifyAssemblyReferencedInDockerfile(
-        new File(getClass().getResource("/docker/Dockerfile_assembly_verify_copy_multiple_valid.test").getPath()),
-        buildConfig, new Properties(),
-        logger);
-
-    // @formatter:off
-    new Verifications() {{
-      logger.warn(anyString, (Object []) any); times = 0;
-    }};
-    //@formatter:on
-  }
-
-  @Test
-  public void testCopyMultipleInvalidVerifyGivenDockerfile(@Injectable final KitLogger logger) throws IOException {
-    BuildConfiguration buildConfig = createBuildConfig().toBuilder()
-        .assembly(AssemblyConfiguration.builder().name("other-layer").build())
-        .build();
-
-    AssemblyManager.verifyAssemblyReferencedInDockerfile(
-        new File(getClass().getResource("/docker/Dockerfile_assembly_verify_copy_valid.test").getPath()),
-        buildConfig, new Properties(),
-        logger);
-
-    // @formatter:off
-    new Verifications() {{
-      logger.warn(anyString, (Object []) any); times = 1;
-    }};
-    //@formatter:on
-  }
 
-  @Test
-  public void testCopyInvalidVerifyGivenDockerfile(@Injectable final KitLogger logger) throws IOException {
-    BuildConfiguration buildConfig = createBuildConfig();
+    @Test
+    public void testCopyOfDockerfile(@Injectable final KitLogger logger) throws IOException, URISyntaxException {
+      BuildConfiguration buildConfig = createBuildConfig();
 
-    AssemblyManager.verifyAssemblyReferencedInDockerfile(
-        new File(getClass().getResource("/docker/Dockerfile_assembly_verify_copy_invalid.test").getPath()),
-        buildConfig, new Properties(),
-        logger);
+      AssemblyManager.verifyAssemblyReferencedInDockerfile(
+              resourceTestFile(dockerFilePath), buildConfig, new Properties(), logger
+      );
 
-    // @formatter:off
-    new Verifications() {{
-      logger.warn(anyString, (Object []) any); times = 1;
-    }};
-    //@formatter:on
+      // @formatter:off
+      new Verifications() {{
+        logger.warn(anyString, (Object []) any); times = expectedNumberOfInvocations;
+      }};
+      //@formatter:on
+    }
   }
 
-  @Test
-  public void testCopyChownValidVerifyGivenDockerfile(@Injectable final KitLogger logger) throws IOException {
-    BuildConfiguration buildConfig = createBuildConfig();
-
-    AssemblyManager.verifyAssemblyReferencedInDockerfile(
-        new File(getClass().getResource("/docker/Dockerfile_assembly_verify_copy_chown_valid.test").getPath()),
-        buildConfig,
-        new Properties(),
-        logger);
-
-    // @formatter:off
-    new Verifications() {{
-      logger.warn(anyString, (Object []) any); times = 0;
-    }};
-    //@formatter:on
+  private static File resourceTestFile(String path) throws URISyntaxException {
```

Comment:
I can have a better look at this. 

The issue I've seen it's that we need the same things in multiple sub-modules.
I usually do a `test-utils` sub-module to collect all such things when I need to reuse them, but I don't know how you guys want to proceed. Maybe adding a new sub-module is not really an option.

Reply:
@sunix please let me know how to proceed here. 
It's indeed duplicated but in different modules. `TestUtils` was added because I needed the same logic in 2 different test classes of the sub-module where it's in.

I would suggest that we merge this as it is and another ticket gets open for refactoring the tests in general.
For example, I've just pulled the `master` branch and the latest commit says something about the tests being converted from JUnit 4 to 5. Unfortunately, only one test was changed :-D